plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ConnectionNameResolver.java

```diff
@@ -0,0 +1,96 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver;
+
+import org.eclipse.jgit.annotations.Nullable;
+import org.jkiss.dbeaver.core.CoreMessages;
+import org.jkiss.dbeaver.model.DBPDataSourceContainer;
+import org.jkiss.dbeaver.model.connection.DBPConnectionConfiguration;
+import org.jkiss.dbeaver.registry.DataSourceDescriptor;
+import org.jkiss.dbeaver.ui.editors.sql.SQLNewScriptTemplateVariablesResolver;
+import org.jkiss.utils.ArrayUtils;
+import org.jkiss.utils.CommonUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class ConnectionNameResolver extends SQLNewScriptTemplateVariablesResolver {
+    public static final String VAR_DEFAULT = "host_or_database";
+    private final DataSourceDescriptor descriptor;
+
+    public static Set<String> getConnectionVariables() {
+        final Set<String> strings = Arrays.stream(ArrayUtils.concatArrays(DBPConnectionConfiguration.ALL_VARIABLES, new String[]{VAR_DEFAULT})).collect(Collectors.toSet());
+        strings.remove(DBPConnectionConfiguration.VARIABLE_DATASOURCE);
+        strings.remove(DBPConnectionConfiguration.VARIABLE_SERVER);
+        return strings;
+    };
+
+    public static List<List<String>> getConnectionVariablesInfo() {
+        final String[][] strings = ArrayUtils.concatArrays(
+            ALL_VARIABLES_INFO,
+            new String[][]{
+                {VAR_DEFAULT, "Legacy name"},
+            });
+        List<List<String>> list = new ArrayList<>();
+        for (String[] string : strings) {
+            if (!string[0].equals(DBPConnectionConfiguration.VARIABLE_DATASOURCE)) {
+                list.add(List.of(string));
+            }
+        }
+        return list;
+    }
+
+    public ConnectionNameResolver(DBPDataSourceContainer dataSourceContainer, DBPConnectionConfiguration configuration, @Nullable DataSourceDescriptor descriptor) {
+        super(dataSourceContainer, configuration);
+        this.descriptor = descriptor;
+    }
+
+    private String generateLegacyConnectionName() {
+        String newName = descriptor == null ? "" : getDataSourceContainer().getName(); //$NON-NLS-1$
+        if (CommonUtils.isEmpty(newName)) {
+            newName = getConfiguration().getDatabaseName();
+            if (CommonUtils.isEmpty(newName) || newName.length() < 3 || CommonUtils.isInt(newName)) {
+                // Database name is too short or not a string
+                newName = getConfiguration().getHostName();
+            }
+            if (CommonUtils.isEmpty(newName)) {
+                newName = getConfiguration().getServerName();
+            }
+            if (CommonUtils.isEmpty(newName)) {
+                newName = getDataSourceContainer().getDriver().getName();
+            }
+            if (CommonUtils.isEmpty(newName)) {
+                newName = CoreMessages.dialog_connection_wizard_final_default_new_connection_name;
+            }
```

Comment:
Maybe it's better to assign variable only if the value is not empty instead of checking the same condition several times?